lib/galaxy/config/sample/galaxy.yml.sample

```diff
@@ -197,8 +197,21 @@ galaxy:
   # Tool config files, defines what tools are available in Galaxy. Tools
   # can be locally developed or installed from Galaxy tool sheds.
   # (config/tool_conf.xml.sample will be used if left unset and
-  # config/tool_conf.xml does not exist).
-  #tool_config_file: config/tool_conf.xml,config/shed_tool_conf.xml
+  # config/tool_conf.xml does not exist). Can be a single file, a list
+  # of files, or (for backwards compatibility) a comma-separated list of
+  # files.
+  #tool_config_file: config/tool_conf.xml
```

Comment:
What if 'config_dir' is set by the admin to something other than 'config'? It would affect this, right? If so, maybe set the default to 'tool_conf.xml', and add to the desc something like 'Default value will be resolved to 'config/tool_conf.xml' where 'config' is the default value of the 'config_dir' option' ? (I haven't fully grokked the processing of these config options, so it's a question, not a suggestion.)

Reply:
Yeah, I think maybe up at the top of the `galaxy` section in the sample we should put a header that explains how relative paths in the file work, like "all paths are relative to the directory in which galaxy.yml lives unless otherwise stated." I actually need to double check that everything works as intended - for example, some paths like the path to `tool-data/` are relative to `galaxy_root` aka `cwd`.

Reply:
I was actually changing some defaults and adding those sort of comments where applicable, but paused when I saw your PR (if it's merged tomorrow, I'd rather wait it out rather than resolving 90 changed lines: I have at least as many for the same file :) 
I think it will be easier to do it per option: some are relative to config/, some to database/, and some, yes, to cwd(). More importantly, some of those defaults *will be used*: since every single  config option 'foo' is now auto-loaded and assigned to 'self.foo', I want to replace all `kwargs.get('foo', bar)` with `self.foo`. So, stuff like `x = resolve_path(kwargs.get('foo', bar), some-path` would become `x = resolve_path(self.foo, some-path)`. Of course, this depends on what is 'bar', and how different it is from the default specified in `config_schema.yml`. Which is why if a default is specified as `config/foo`, doing the above would result in `config/config/foo`. And bad things would happen.

Reply:
> I think it will be easier to do it per option: some are relative to config/, some to database/, and some, yes, to cwd(). 

Yeah, ok, this makes sense. Sorry, I had a quick look over your config standardization PR last night but haven't had a chance to grok it.

We might even want to make a standard for this, like a new option in the schema (`path_relative_to`?) with values being other config vars (e.g. `data_dir`) or `cwd`, and then those can be automatically added to the comment just above the var in the sample?

Also, anything relative to `cwd` was supposed to be eliminated in #921, but not all of that was complete when it was merged.

Reply:
`path_relative_to` - I think this is a very good idea! I'll play around with it..
`cwd`: I won't touch those for now; once the rest mostly done we can address these on a case by case basis.

Reply:
It looks like I missed the mark in #921, if you look at [_parse_config_file_options](https://github.com/galaxyproject/galaxy/blob/be5aafd34b1f400704ef863684c9b742e1ca7427/lib/galaxy/config/__init__.py#L165) it resolves any configured values relative to `self.root` (aka `cwd`). Only the defaults if the option is unset are relative to other directories.

I'll probably address that in a separate PR though.

Reply:
And those I'm not touching yet because they are in the BaseAppConfiguration, which does not load the schema (although the child can pass the loaded defaults via _set_config_base of course).
Also self.tool_data_path, i think - also resolves to cwd. https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/config/__init__.py#L284